Makefile

```diff
@@ -32,8 +32,8 @@ format:
 	@$(GO) fmt $(pkgs)
 
 $(FIRST_GOPATH)/bin/promu promu:
-	@GOOS=$(shell uname -s | tr A-Z a-z) \
-		GOARCH=$(subst x86_64,amd64,$(patsubst i%86,386,$(patsubst arm%,arm,$(shell uname -m)))) \
+	@GOOS=$(go env GOOS) \
+		GOARCH=$(go env GOARCH) \
```

Comment:
Unfortunately, I think setting these will break cross-building.
When cross-building we explicitly set those envvars to the target platform. Since this target is called when using the `build` target, it will build for the target cross-build platform, instead of the builder platform, which will make `promu` unusable (prometheus/golang-builder images).

Reply:
```
% go env GOOS
darwin
% env GOOS=solaris go env GOOS
solaris
% env GOOS=FreeBSD go env GOOS
FreeBSD
% env GOOS=freebsd go env GOOS
freebsd
% go env GOARCH
amd64
% env GOARCH=amd64 go env GOARCH
amd64
% env GOARCH=doesnt_exist go env GOARCH
doesnt_exist
```

Reply:
Interesting! We should fix this in all our Makefiles then.

Reply:
It won't work with the current mechanism when crossbuilding.

If we take the following Makefile:

```Makefile
$(FIRST_GOPATH)/bin/promu promu:
	@GOOS=$($(GO) env GOOS) echo $(GOOS)
	@GOARCH=$($(GO) env GOARCH) echo $(GOARCH)
	@GOOS=$($(GO) env GOOS) \
		GOARCH=$($(GO) env GOARCH) \
		$(GO) get -u github.com/prometheus/promu


build: $(PROMU)
	@echo ">> building binaries"
        @$(PROMU) build --prefix $(PREFIX)
```

and given that golang-builder docker image call the Makefile `build` target like this:

```bash
# goos=solaris
# goarch=amd64
GOOS=${goos} GOARCH=${arch} make build
```

The result will be a `go get promu` for the `solaris/adm64` platform instead of the `linux/amd64` platform of the docker image.

Reply:
@sdurrheimer So `go env GOHOSTOS` and `go env GOHOSTARCH` would do what we want then?

Reply:
@grobie wow, didn't know about those envvars, nice !

Reply:
@sean- Can you update you're PR by using `GOHOSTOS` and `GOHOSTARCH` envvars ?

Reply:
Done!